common/persistence/src/main/java/app/coronawarn/server/common/persistence/repository/FederationUploadKeyRepository.java

```diff
@@ -0,0 +1,34 @@
+/*-
+ * ---license-start
+ * Corona-Warn-App
+ * ---
+ * Copyright (C) 2020 SAP SE and all other contributors
+ * ---
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ---license-end
+ */
+
+package app.coronawarn.server.common.persistence.repository;
+
+import app.coronawarn.server.common.persistence.domain.DiagnosisKey;
+import java.util.Collection;
+import org.springframework.data.jdbc.repository.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface FederationUploadKeyRepository
+    extends org.springframework.data.repository.Repository<DiagnosisKey, Long> {
```

Comment:
The idea is that we use the DiagnosisKey class here just to be able to deserialize uploadable keys. The actual table is referenced in the native query. 
 If we were to switch to a CrudRepository, it would allow manipulation of the actual 'daignosis_key' table via generated methods .That's because spring repos use the entity metadata to find the concrete table.

Reply:
Ok, fine for me